crates/ty_ide/src/completion.rs

```diff
@@ -887,9 +890,16 @@ fn is_in_definition_place(db: &dyn Db, tokens: &[Token], file: File) -> bool {
 /// This has the effect of putting all dunder attributes after "normal"
 /// attributes, and all single-underscore attributes after dunder attributes.
 fn compare_suggestions(c1: &Completion, c2: &Completion) -> Ordering {
-    let (kind1, kind2) = (NameKind::classify(&c1.name), NameKind::classify(&c2.name));
+    fn key<'a>(completion: &'a Completion) -> (Option<&'a ModuleName>, NameKind, bool, &'a Name) {
+        (
+            completion.module_name,
```

Comment:
Hmmmm. This is sorting by module name rather than the _existence_ of module name. I think we might just want `completion.module_name.is_none()` here? (That should make the `None` case sort before any `Some` case.)